test/module/irohad/ametsuchi/block_query_test.cpp

```diff
@@ -214,3 +217,146 @@ TEST_F(BlockQueryTest, GetTransactionsWithInvalidTxAndValidTx) {
   });
   ASSERT_TRUE(wrapper.validate());
 }
+
+/**
+ * @given block store with 2 blocks totally containing 3 txs created by
+ * user1@test AND 1 tx created by user2@test
+ * @when get non-existent 1000th block
+ * @then nothing is returned
+ */
+TEST_F(BlockQueryTest, GetNonExistentBlock) {
+  auto wrapper = make_test_subscriber<CallExact>(
+      blocks->getBlocks(1000, 1), 0);
+  wrapper.subscribe();
+  ASSERT_TRUE(wrapper.validate());
+}
+
+/**
+ * @given block store with 2 blocks totally containing 3 txs created by
+ * user1@test AND 1 tx created by user2@test
+ * @when height=1, count=1
+ * @then returned exactly 1 block
+ *
+ * @note test for bug, when <total_blocks=2, height=1, count=1>, then
```

Comment:
Redundant info about fixed bug in the code.

Reply:
It is not redundant. Next developer may change the code and get the same bug, then he/she needs to understand what caused that bug.

Reply:
Then there should be a comment about calculation in the following code, so that other developer will not change the code, but not in the test:
https://github.com/hyperledger/iroha/blob/97799acf1fc72c8e423dc8f36fe37f64f9f61c12/irohad/ametsuchi/impl/redis_block_query.cpp#L30-L34